source/core/lighting/photons.cpp

```diff
@@ -2280,18 +2280,21 @@ void PhotonGatherer::gatherPhotonsRec(int start, int end)
 
     // check this photon
 
-    // find distance from pt
-    ptToPhoton = - (*pt_s) + Vector3d(photon->Loc);
-    // all distances are squared
-    dx = ptToPhoton[X]*ptToPhoton[X];
-    dy = ptToPhoton[Y]*ptToPhoton[Y];
-    dz = ptToPhoton[Z]*ptToPhoton[Z];
+    // find distance in DimToUse from pt
+    d = photon->Loc[DimToUse] - (*pt_s)[DimToUse];
+    d *= d;
 
-    if (!(  ((dx>dmax_s) && ((DimToUse)==X)) ||
-            ((dy>dmax_s) && ((DimToUse)==Y)) ||
-            ((dz>dmax_s) && ((DimToUse)==Z)) ))
+    if (d<dmax_s)
     {
-        // it fits manhatten distance - maybe we can use this photon
+        // it fits manhatten, DimToUse distance - maybe we can use this photon
+
+        dx = photon->Loc[X] - (*pt_s)[X];
+        dy = photon->Loc[Y] - (*pt_s)[Y];
+        dz = photon->Loc[Z] - (*pt_s)[Z];
+        ptToPhoton = Vector3d(dx,dy,dz);
+        dx *= dx;
+        dy *= dy;
+        dz *= dz;
```

Comment:
Your suggestions worked out great! Better performance than mine - I think due eliminating the duplicate delta calculation.  I'll update this pull request against master shortly. 

Performance data for reference: 
----------------------

_master  - Current master.
_gphr    - My proposed updates.
_gphrCL  - Your improvements.


povrayGNU_master   optics.pov +W800 +H600 +A0.1 +am2 +r3 +wt2 -D -P -cc
-----------------------------------------------------------------------
227.45user 0.04system 1:54.76elapsed

povrayCLANG_master optics.pov +W800 +H600 +A0.1 +am2 +r3 +wt2 -D -P -cc
-----------------------------------------------------------------------
215.62user 0.04system 1:48.87elapsed



povrayGNU_gphr     optics.pov +W800 +H600 +A0.1 +am2 +r3 +wt2 -D -P -cc
-----------------------------------------------------------------------
199.88user 0.06system 1:40.96elapsed -12.12%

povrayCLANG_gphr   optics.pov +W800 +H600 +A0.1 +am2 +r3 +wt2 -D -P -cc
-----------------------------------------------------------------------
197.34user 0.05system 1:39.66elapsed -8.48%



povrayGNU_gphrCL   optics.pov +W800 +H600 +A0.1 +am2 +r3 +wt2 -D -P -cc
-----------------------------------------------------------------------
227.13user 0.07system 1:54.79elapsed  (Update process kicked off - cache degrade)
196.67user 0.03system 1:39.30elapsed
196.74user 0.01system 1:39.31elapsed  -13.52% (-1.59% over my version)

povrayCLANG_gphrCL  optics.pov +W800 +H600 +A0.1 +am2 +r3 +wt2 -D -P -cc
-----------------------------------------------------------------------
192.48user 0.02system 1:37.26elapsed
192.42user 0.04system 1:37.26elapsed  -10.75% (-2.48% over my version)